asdf/yamlutil.py

```diff
@@ -318,11 +324,11 @@ def dump_tree(tree, fd, ctx):
 
     ctx : Context
         The writing context.
-    """
-    class AsdfDumperTmp(AsdfDumper):
-        pass
-    AsdfDumperTmp.ctx = ctx
 
+    tree_finalizer : callable, optional
+        Callback that receives the tagged tree before it is validated
+        and defaults are removed.
```

Comment:
It receives the tagged tree, does something with it, then what?  I.e. what is the purpose of this hook?

Reply:
I'll add a note explaining what the library does with it.